demos/python_demos/image_embeddings/Readme.md

```diff
@@ -0,0 +1,219 @@
+   - Extracts embeddings using the OpenVINO-served models.
+   - Stores embeddings in a **Vector Database** (Qdrant)
+
+4. **Search API (`search_images.py`)**
+   - Accepts an input query image.
+   - Generates its embedding and queries the Vector DB.
+   - Returns the most similar images based on cosine similarity or other distance metrics.
+
+5. **Search App (`streamlit_app.py`)**
+   - Provided a frontend for the users to interact with the project and test it
+   - Allows users to upload images and perform semantic search
+
+
+## Installation and setup
+
+```bash
+cd demos/python_demos/image_embeddings
+python3 -m venv venv
+pip install -r requirements.txt
+```
+
+
+## Model conversion
+```bash
+python3 -m venv venv
+cd demos/python_demos/image_embeddings/model_conversion
+```
+
+For clip
+```bash
+python clip_conversion.py
+```
+
+For laion
+```bash
+python laion_conversion.py
+```
+
+For Dino
+```bash
+python dino_conversion.py
+```
+
+## Deploying OpenVINO Model Server
+Prerequisites:
+-  image of OVMS with Python support and Optimum installed
+Mount the `./servable` which contains:
+- `post.py` and `pre.py` - python scripts which are required for execution.
+- `config_model.json` - which defines which servables should be loaded.
+- `graph_clip.pbtxt`, `graph_laion.pbtxt`, `graph_dino.pbtxt` - which defines MediaPipe graph containing python nodes.
+
+
+```bash
+cd demos/python_demos/image_embeddings
+```
+
+
+To use CPU
+```bash
+docker run -it --rm \
+-p 9000:9000 -p 8000:8000 \
+-v ${PWD}/servable:/workspace \
+-v ${PWD}/model_conversion/saved_mod/siglip:/saved_mod/dino \
+-v ${PWD}/model_conversion/saved_mod/clip:/saved_mod/clip \
+-v ${PWD}/model_conversion/saved_mod/laion:/saved_mod/laion \
+openvino/model_server:py \
+--config_path /workspace/config_model.json \
+--port 9000 --rest_port 8000
+
+```
+
+To use GPU
+```bash
+docker run -it --rm \
+  --device=/dev/dxg \
+  --volume /usr/lib/wsl:/usr/lib/wsl \
+  -p 9000:9000 -p 8000:8000 \
+  -v ${PWD}/servable:/workspace \
+  -v ${PWD}/model_conversion/saved_mod/dino:/saved_mod/dino \
+  -v ${PWD}/model_conversion/saved_mod/clip:/saved_mod/clip \
+  -v ${PWD}/model_conversion/saved_mod/laion:/saved_mod/laion \
+  ovms-gpu-custom \
+    --config_path /workspace/config_model.json \
+    --port 9000 \
+    --rest_port 8000
+```
+
+## Deploying the Vector Database
+
+The next step is to start the vector database. In this project, we are using **Qdrant**, an open-source vector database optimized for efficient semantic search.
+
+Run the following command to start Qdrant with Docker:
+
+```bash
+cd demos/python_demos/image_embeddings
+docker run -p 6333:6333 qdrant/qdrant
+```
+
+## Running the Demo
+
+Once the vector database is running and the OpenVINO Model Server is deployed, open another terminal and run:
+
+```bash
+source venv/bin/activate
+cd demos/python_demos/image_embeddings
+python grpc_cli.py
+or
+python grpc_cli.py --model "your selected model name"
+```
```

Comment:
Use actual code in bash script - select one way and stick to it, select real model, but you can explain in text bottom that there are other options as well

all bash scripts run in our tests so they should be legit